src/cargo/ops/vendor.rs

```diff
@@ -497,8 +497,8 @@ fn prepare_for_vendor(
     packaged_files: &[PathBuf],
     gctx: &GlobalContext,
 ) -> CargoResult<Package> {
-    let contents = me.manifest().contents();
-    let document = me.manifest().document();
+    let contents = me.manifest().contents().unwrap();
```

Comment:
My default assumption is that this PR is not ready to merge until all `.unwrap()`s are resolved.  Any remaining `.unwrap()` should have a justification, likely including switching it to an `.expect()` explaining what invariants are keeping it from panicking.